src/api/server.rs

```diff
@@ -0,0 +1,123 @@
+use super::{handler, schema::build_schema};
+
+use async_graphql::{
+    http::{playground_source, GraphQLPlaygroundConfig},
+    QueryBuilder,
+};
+use async_graphql_warp::{graphql_subscription, GQLResponse};
+use std::convert::Infallible;
+use std::net::SocketAddr;
+use tokio::sync::oneshot;
+use tokio::sync::oneshot::{Receiver, Sender};
+use warp::filters::BoxedFilter;
+use warp::{http::Response, Filter, Reply};
+
+pub struct Server {
+    /// Address for the API server to bind on
+    address: SocketAddr,
+
+    /// Transmission channel to trigger closure of a running API server
+    trigger_cancel: Sender<()>,
+
+    /// Receiver signal to cancel running API server
+    cancel_signal: Option<Receiver<()>>,
+}
+
+impl Server {
+    /// Returns a new API Server
+    pub fn new(address: SocketAddr) -> Server {
```

Comment:
```suggestion
    pub fn new(address: SocketAddr) -> Self {
```

(not sure how strictly Clippy is configured in Vector, but might be good to run Clippy locally as well, as it should catch cases like these)